FreeRTOS/FreeRTOS
# Adding CMakeList.txt in posix GCC demos

Description
-----------
* Add CMakeList.txt in posix GCC demos to consume FreeRTOS-Kernel/CMakeLists.txt
* Add the following option to match existing Makefile

**Trace on enter**
```
cmake -S . -B build -DTRACE_ON_ENTER=1
```
**Coverage test**
```
cmake -S . -B build -DCOVERAGE_TEST=1
```
**Profile**
```
cmake -S . -B build -DPROFILE=1
```
**sanitize address and alignment**
```
cmake -S . -B build -DSANITIZE_ADDRESS=1
```
**sanitize leak**
```
cmake -S . -B build -DSANITIZE_LEAK=1
```

Test Steps
-----------
```
cd ./FreeRTOS/FreeRTOS/Demo/Posix_GCC
cmake -B build -S .
cd build
make
```
To build with clang compiler, you will need to suppress some compiler warning
```
target_compile_options( freertos_kernel
  PRIVATE
    $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-cast-align>
    $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-covered-switch-default>
    $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-missing-variable-declarations>
    $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-padded>
    $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-unused-macros>
)
target_compile_options( freertos_kernel_port
  PRIVATE
     $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-disabled-macro-expansion>
     $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-padded>
     $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-reserved-id-macro>
)
```

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
